src/test.rs

```diff
@@ -543,6 +547,25 @@ impl<S: 'static> TestRequest<S> {
         self
     }
 
+    /// set cookie of this request
+    pub fn cookie(mut self, cookie: Cookie<'static>) -> Self {
+        let mut should_insert = true;
+        match &mut self.cookies {
+            Some(old_cookies) => {
+                for old_cookie in old_cookies.iter() {
+                    if old_cookie == &cookie {
+                        should_insert = false
+                    };
+                };
+                if should_insert {
+                    old_cookies.push(cookie);
+                };
+            }
+            None => {}
```

Comment:
But we need to insert cookie, if there is `None` in `self.cookies` right?

Reply:
@DoumanAsh Would you please review again ? It's done.